### PR TITLE
Re-structure the docs into sections

### PR DESCRIPTION
**Pull request by nolar**
## What do these changes do?

Re-structures the documentation into multiple sections, and extracts some common topics from the Handlers page to their own (kwargs, error handling, filtering, etc).

## Description

As new types of handlers and decorators are being added (e.g. activities), the docs were growing and eventually ended up being overloaded with informations. As even more handlers are going to appear soon (e.g. daemons & timers), it requires re-structuring the docs, specifically extracting some common topics as separate pages.

Some new docs are also written: slightly more verbose on kwargs, totally new for results delivery.

A preview is temporarily available at:

* https://kopf-nolars-fork.readthedocs.io/en/doc-common-aspects-as-pages/index.html

## Issues/PRs

> Issues:  #12 #19 

## Type of changes

- Mostly documentation and examples (no code changes)

## Checklist

- [x] The code addresses only the mentioned problem, and this problem only
- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`

### CODE REVIEW

File: docs/results.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
+================
+Results delivery
+================
+
+All handlers can result arbitrary JSON-serializable values.
```
Comment (haikoschol) [inline docs/results.rst]: `s/result/return/`?

File: docs/kwargs.rst [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+.. kwarg:: reason
+
+Causation
+---------
+
+**DEPRECATED:**
+``cause`` is the processed cause of the handler as detected by the framework (create/update/delete).
+
+``reason`` is a type of change detection (creation, update, deletion, resuming).
+It is generally reflected in the handler decorator used , but can be useful for
```
Comment (haikoschol) [inline docs/kwargs.rst]: nit-est of nits: There's a whitespace in front of the comma.

File: docs/results.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+.. code-block:: none
+
+    ...
+    status:
+      create_kex_1: 100
+      create_kex_2:
+        r1: 66
+        r2: 666
+
+The returning results can be used to communicate between handlers through
```
Comment (haikoschol) [inline docs/results.rst]: Maybe use "returned values" instead of "returning results".

Comment (nolar) [reply docs/results.rst]: "function results" sounds better. Fixed.

File: docs/kwargs.rst (hunk 2 above)
Comment (nolar) [reply docs/kwargs.rst]: Fixed

File: docs/results.rst (hunk 1 above)
Comment (nolar) [reply docs/results.rst]: Fixed

**Pull request merged**